### PR TITLE
[RF] Fix recently-introduced `RooAbsReal::getPropagatedError` problems and add several new unit tests

### PR DESCRIPTION
**Pull request by guitargeek**
The recent commit `565e9aa19a` updated the `getPropagatedError` with
some additional checks, but these checks caused segfaults in some
specifit usecases:

* where the `RooAbsReal` is a parameter in the fit result itself

* where the `RooAbsReal` is a variable unrelated to the fit result

* where the `RooAbsReal` depends only on some of the parameters in the
  fit result

This commit fixes these usecases again.

A unit test that checks that these usecases don't fail anymore is now
implemented in `testRooAbsReal`.

This PR also adds some other new unit tests that cover recent PRs and issues.

### CODE REVIEW

**Comment by guitargeek** [comment]
The last push didn't change much, just commenting out the unused variables

File: roofit/roofitcore/src/RooAbsReal.cxx [hunk 1]
```diff
@@ -2660,9 +2673,9 @@ double RooAbsReal::getPropagatedError(const RooFitResult &fr, const RooArgSet &n
   TMatrixDSym C(paramList.getSize()) ;
   vector<double> errVec(paramList.getSize()) ;
```
Comment (lmoneta) [inline roofit/roofitcore/src/RooAbsReal.cxx:2674]: here better using std::vector

File: roofit/roofitcore/src/RooAbsReal.cxx [hunk 2]
```diff
@@ -2605,23 +2605,36 @@ double RooAbsReal::getPropagatedError(const RooFitResult &fr, const RooArgSet &n
...
+  for(auto * rrvFitRes : static_range_cast<RooRealVar*>(fr.floatParsFinal())) {
+
+     auto rrvInAbsReal = static_cast<RooRealVar const*>(allParamsInAbsReal.find(*rrvFitRes));
+
+     // If this RooAbsReal is a RooRealVar in the fit result, we don't need to
+     // propagate anything and can just return the error in the fit result
+     if(rrvFitRes->namePtr() == namePtr()) return rrvFitRes->getError();
+
+     // Strip out parameters with zero error
+     if (rrvFitRes->getError() <= 1e-20) continue;
```
Comment (lmoneta) [inline roofit/roofitcore/src/RooAbsReal.cxx:2618]: why hard coding here 1.E-20 ? Maybe use numerical precision epsilon ?

**Review by lmoneta — APPROVED** [review]
LGTM!
Thanks for the fix, I have just a couple of small comments

File: roofit/roofitcore/src/RooAbsReal.cxx (hunk 1 above)
Comment (guitargeek) [reply roofit/roofitcore/src/RooAbsReal.cxx]: Okay! I have done the same for all `std::` classes in the file

File: roofit/roofitcore/src/RooAbsReal.cxx (hunk 2 above)
Comment (guitargeek) [reply roofit/roofitcore/src/RooAbsReal.cxx]: Yes, that is indeed a better choice than this magic number. I have updated it here and also in another place in that file

**Pull request merged**